### PR TITLE
Add the ability to use the LED indicators for CAPS_LOCK et al.

### PR DESCRIPTION
**Pull request by jfurcean**
This adds the ability to read the LED status of `CAPS_LOCK`, `NUM_LOCK`, `SCROLL_LOCK`, & `COMPOSE`.

```py
LED_NUM_LOCK  # Works on Windows, RaspberryPi
LED_CAPS_LOCK # Works on Windows, RaspberryPi, Mac
LED_SCROLL_LOCK # Works on Windows
LED_COMPOSE  # Not Tested
```

## Test Code
```py
import usb_hid
from adafruit_hid.keyboard import Keyboard
from adafruit_hid.keycode import Keycode
import time

KEY_DELAY = .09

key_actions = [
    {
        "name": "Caps Lock",
        "keycode": Keycode.CAPS_LOCK,
        "led_code": Keyboard.LED_CAPS_LOCK
    },
    {
        "name": "Caps Lock",
        "keycode": Keycode.SCROLL_LOCK,
        "led_code": Keyboard.LED_SCROLL_LOCK
    },
    {
        "name": "Caps Lock",
        "keycode": Keycode.KEYPAD_NUMLOCK,
        "led_code": Keyboard.LED_NUM_LOCK
    },
]
kbd = Keyboard(usb_hid.devices)

for action in key_actions:
    for i in range(2):
        kbd.press(action['keycode'])
        time.sleep(KEY_DELAY)
        kbd.release(action['keycode'])
        print(f"{action['name']}: {kbd.led_on(action['led_code']}"))
    print()
```

### CODE REVIEW

File: adafruit_hid/keyboard.py [hunk 1]
```diff
@@ -22,6 +22,11 @@
 class Keyboard:
     """Send HID keyboard reports."""
 
+    LED_NUM_LOCK = 0x01
+    LED_CAPS_LOCK = 0x02
+    LED_SCROLL_LOCK = 0x04
+    LED_COMPOSE = 0x08
```
Comment (dhalbert) [inline adafruit_hid/keyboard.py:28]: Please add documentation for each of these as a docstring after the assignment.

File: adafruit_hid/keyboard.py [hunk 2]
```diff
@@ -143,3 +148,12 @@ def _remove_keycode_from_report(self, keycode):
...
                 if self.report_keys[i] == keycode:
                     self.report_keys[i] = 0
+
+    @property
+    def led_status(self):
+        """Returns the last received report"""
+        return self._keyboard_device.last_received_report
+
+    def led_on(self, led_code):
+        """Returns whether an LED is on based on the led code"""
```
Comment (dhalbert) [inline adafruit_hid/keyboard.py:158]: Could you include an example, using one or more of the constants?

**Review by dhalbert — CHANGES_REQUESTED** [review]
Thanks for doing this! It is simple and small. A little more documentation is all that is needed.

File: adafruit_hid/keyboard.py (hunk 1 above)
Comment (jfurcean) [reply adafruit_hid/keyboard.py]: I have added docstrings after each assignment.

File: adafruit_hid/keyboard.py (hunk 2 above)
Comment (jfurcean) [reply adafruit_hid/keyboard.py]: I have added an example using LED_CAPS_LOCK.

**Review by dhalbert — APPROVED** [review]
Thank you for this valuable addition to the library!

**Comment by nico7885** [comment]
@dhalbert,
Ok, thank you very much, good news.

**Pull request merged**